sphinx/util/osutil.py

```diff
@@ -238,17 +236,13 @@ class FileAvoidWrite:
     def __init__(self, path):
         # type: (str) -> None
         self._path = path
-        self._io = None  # type: Union[StringIO, BytesIO]
+        self._io = None  # type: StringIO
 
     def write(self, data):
```

Comment:
To make sure `data` is an instance of str, please add `assert` statement here.

Reply:
The assert would be redundant with Python 3 StringIO's own type checking. Observe:

```py
>>> import io
>>> buf = io.StringIO()
>>> buf.write(b'')
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
TypeError: string argument expected, got 'bytes'
```

WDYT?

Reply:
You're right. No assertion is needed.